### ISSUE TITLE
extrapolated galaxy models

### ISSUE THREAD

**Issue by rmandelb** [GalSim-developers/GalSim#808]
For HSC simulations, Francois ( @EiffL ) has been working on a way to easily extrapolate the COSMOS I<25.2 sample to fainter magnitudes and make a deeper parametric galaxy sample.  Doing this extrapolation isn't likely good enough for precise tests of shear calibration for deeper samples, but it's likely good enough to just test for issues like "my real galaxies brighter than 25.2 are blended with faint galaxies some fraction of the time; what's the impact of that?"  My question is whether there's interest in having a routine that will make synthetic extrapolated Sersic samples in GalSim?  If there is interest then Francois would be happy to incorporate his code and make a routine for this in scene.py.

**Comment by rmjarvis** [comment]
I don't think anyone has as pressing a need for this as you guys do.  :)  

But I expect we'll probably want something like that for LSST and WFirst tests down the road, so putting it here would indeed be appreciated.  You might ping the LSST DESC list to see if anyone has already taken a stab at doing this, since I think some people have probably tried to look at distributions of galaxies that go fainter than COSMOS.  Especially for blending studies.

**Comment by rmandelb** [comment]
Good point.  My understanding had been that everyone who has done that (e.g., @dkirkby ) have been using the PhoSim input catalogs if they want to have a deep simulated galaxy sample.  @dkirkby , do you know of other strategies in use?

My intention here was to have a lighter-weight scheme with fewer dependencies (i.e., just the samples that GalSim already knows about).

**Comment by rmandelb** [comment]
Note that this code is available at https://github.com/EiffL/cosmos_extrapolation , with an ipython notebook demonstrating the strategy.  It's not integrated into GalSim yet, but this gives an idea of how it works.  @esheldon - you might want to check this out?

**Comment by dkirkby** [comment]
This looks like a nice approach.  The method we use in the DESC [WeakLensingDeblending package](http://weaklensingdeblending.readthedocs.io/en/latest/) is to read the LSST simulated catalog ("CatSim").  Working directly with the catalog brings in a lot of LSST dependencies, but the package can also read a 1 sq.deg. FITS file extracted from the catalog (documented [here](http://weaklensingdeblending.readthedocs.io/en/latest/catalog.html)) without bringing in any LSST dependencies.  It would be interesting to compare CatSim distributions with these extrapolations.

**Comment by rmandelb** [comment]
Thanks, David!  I forgot about the option to read in a FITS file without bringing in all the many dependencies.  I agree that a comparison would be interesting!

**Comment by esheldon** [comment]
I would be interested to see the extrapolation plots for the
half-light-radius as well, and even more interesting the joint
distribution of flux and size.  This space has multiple populations,
complicating the extrapolation.

I've also been using a KDE to sample the parameter space, seems to
work well.  I ended up setting the "kde factor" by hand to avoid
over-smoothing.

**Comment by dkirkby** [comment]
KDE allows you to smooth and interpolate, but not extrapolate, right?

**Comment by esheldon** [comment]
On 10/19/16, David Kirkby <notifications@github.com> wrote:
> KDE allows you to smooth and interpolate, but not extrapolate, right?

right, I have not done any extrapolation my self, was just confirming
I had seen kde has worked well for sampling the flux-hlr space.


-- 
Erin Scott Sheldon
Brookhaven National Laboratory erin dot sheldon at gmail dot com
